plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java

```diff
@@ -0,0 +1,165 @@
+                    TiDBMessages.tidb_connection_page_label_host);
+            hostLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            hostText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.FILL_HORIZONTAL);
+            gd.grabExcessHorizontalSpace = true;
+            hostText.setLayoutData(gd);
+            hostText.addModifyListener(textListener);
+
+            Label portLabel = UIUtils.createControlLabel(hostGroup,
+                    TiDBMessages.tidb_connection_page_label_port);
+            portLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            portText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+            gd.widthHint = UIUtils.getFontHeight(portText) * 5;
+            portText.setLayoutData(gd);
+            portText.addVerifyListener(UIUtils.getIntegerVerifyListener(Locale.getDefault()));
+            portText.addModifyListener(textListener);
+
+            Label databaseLabel = UIUtils.createControlLabel(hostGroup,
+                    TiDBMessages.tidb_connection_page_label_database);
+            databaseLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            databaseText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.FILL_HORIZONTAL);
+            gd.grabExcessHorizontalSpace = true;
+            gd.horizontalSpan = 3;
+            databaseText.setLayoutData(gd);
+            databaseText.addModifyListener(textListener);
+        }
+
+        createAuthPanel(addrGroup, 1);
+        createDriverPanel(addrGroup);
+        setControl(addrGroup);
+    }
+
+    @Override
+    public boolean isComplete() {
+        return super.isComplete() && hostText != null && databaseText != null && portText != null
+                && !CommonUtils.isEmpty(hostText.getText());
+    }
+
+    @Override
+    public void loadSettings() {
+        // Load values from new connection info
+        DBPDataSourceContainer activeDataSource = site.getActiveDataSource();
+        DBPConnectionConfiguration connectionInfo = activeDataSource.getConnectionConfiguration();
+        if (!CommonUtils.isEmpty(connectionInfo.getHostName())) {
+            hostText.setText(CommonUtils.notEmpty(connectionInfo.getHostName()));
+        } else {
+            hostText.setText(MySQLConstants.DEFAULT_HOST);
+        }
+        if (!CommonUtils.isEmpty(connectionInfo.getHostPort())) {
+            portText.setText(CommonUtils.notEmpty(connectionInfo.getHostPort()));
+        } else if (site.getDriver().getDefaultPort() != null) {
+            portText.setText(site.getDriver().getDefaultPort());
+        } else {
+            portText.setText("");
+        }
+        if (!CommonUtils.isEmpty(connectionInfo.getDatabaseName())) {
+            databaseText.setText(CommonUtils.notEmpty(connectionInfo.getDatabaseName()));

```

Comment:
Please remove `CommonUtils.notEmpty`. You already checked it with `!CommonUtils.isEmpty`